### PR TITLE
fix: not receiving messages on second account [AR-2867]

### PR DESCRIPTION
**Pull request by Garzas**
<table>
<td>
  <a href="https://wearezeta.atlassian.net/browse/AR-2867" title="AR-2867" target="_blank">BugAR-2867</a>  Notifications don't show all notifications
  </td></table>
  <br />
 

----
#### PR Submission Checklist for internal contributors

- The **PR Title**
    - [x] conforms to the style of semantic commits messages¹ supported in Wire's Github Workflow²
    - [x] contains a reference JIRA issue number like `SQPIT-764`
    - [x] answers the question: _If merged, this PR will: ..._ ³

- The **PR Description**
    - [ ] is free of optional paragraphs and you have filled the relevant parts to the best of your ability
----

# What's new in this PR?

### Issues

When app is in foreground and we are logged to 2 accounts not current account doesn't receive all notifications

### Solutions

Disable downgrading connection policy by not current account

Needs releases with:

- [ ] GitHub link to other pull request

#### How to Test
1. Login in one device to 2 accounts
2. Create group with 3 users
3. Leave app in foreground
4. Type fast few messages from third user on web to group conversation
5. Notifications should contain same messages for both users

### Notes (Optional)

_Specify here any other facts that you think are important for this issue._

### Attachments (Optional)

_Attachments like images, videos, etc. (drag and drop in the text box)_ 

----
#### PR Post Submission Checklist for internal contributors (Optional)

- [ ] Wire's Github Workflow has automatically linked the PR to a JIRA issue
----
#### PR Post Merge Checklist for internal contributors

- [ ] If any soft of configuration variable was introduced by this PR, it has been added to the relevant documents and the CI jobs have been updated.
----
##### References
1. https://sparkbox.com/foundry/semantic_commit_messages
1. https://github.com/wireapp/.github#usage
1. E.g. `feat(conversation-list): Sort conversations by most emojis in the title #SQPIT-764`.

### CODE REVIEW

**Comment by AndroidBob** [comment]
Build [2](https://10.10.124.134/job/wire-android-reloaded/job/PR-1438/2/) **failed**.

**Comment by AndroidBob** [comment]
Build [3](https://10.10.124.134/job/wire-android-reloaded/job/PR-1438/3/) **failed**.

**Comment by AndroidBob** [comment]
Build [4](https://10.10.124.134/job/wire-android-reloaded/job/PR-1438/4/) **succeeded**.

The build produced the following APK's:
- [com.wire.android-vv4.0.0(5698148)(5698148)-staging-compat.apk](https://z-lohika.s3-eu-west-1.amazonaws.com/megazord/android/reloaded/staging/compat/com.wire.android-vv4.0.0(5698148)(5698148)-staging-compat.apk)

**Comment by AndroidBob** [comment]
Build [5](https://10.10.124.134/job/wire-android-reloaded/job/PR-1438/5/) **succeeded**.

The build produced the following APK's:
- [com.wire.android-vv4.0.0(5699666)(5699666)-staging-compat.apk](https://z-lohika.s3-eu-west-1.amazonaws.com/megazord/android/reloaded/staging/compat/com.wire.android-vv4.0.0(5699666)(5699666)-staging-compat.apk)

File: app/src/main/kotlin/com/wire/android/util/lifecycle/ConnectionPolicyManager.kt [hunk 1]
```diff
@@ -125,39 +124,25 @@ class ConnectionPolicyManager @Inject constructor(
     private suspend fun UserSessionScope.downgradePolicyIfNeeded(
         userId: UserId
     ) {
-        val isCurrentSession = isCurrentSession(userId)
         val isAppOnForeground = currentScreenManager.isAppOnForegroundFlow().first()
-        logger.d("isCurrentSession = $isCurrentSession; hasInitialisedUI = $isCurrentSession")
-        val shouldKeepLivePolicy = isCurrentSession && isAppOnForeground
-        if (!shouldKeepLivePolicy) {
-            logger.d("$TAG Downgrading policy as conditions to KEEP_ALIVE are not met")
+        logger.d("$TAG hasInitialisedUI = $isAppOnForeground")
```
Comment (alexandreferris) [inline app/src/main/kotlin/com/wire/android/util/lifecycle/ConnectionPolicyManager.kt]:
```suggestion
        logger.d("$TAG isAppOnForeground = $isAppOnForeground")
```

File: app/src/main/kotlin/com/wire/android/util/lifecycle/ConnectionPolicyManager.kt [hunk 2]
```diff
@@ -125,39 +124,25 @@ class ConnectionPolicyManager @Inject constructor(
...
+        if (!isAppOnForeground) {
+            logger.d("$TAG $userId Downgrading policy as conditions to KEEP_ALIVE are not met")
```
Comment (alexandreferris) [inline app/src/main/kotlin/com/wire/android/util/lifecycle/ConnectionPolicyManager.kt]: Is it possible to obfuscate `$userId` here?

**Review by alexandreferris — APPROVED** [review]
Great work! I just have two small comments, but nothing much urgent

File: app/src/main/kotlin/com/wire/android/util/lifecycle/ConnectionPolicyManager.kt (hunk 2 above)
Comment (Garzas) [reply app/src/main/kotlin/com/wire/android/util/lifecycle/ConnectionPolicyManager.kt]: Yup I forgot to hide it after debugging 🤦

**Review by borichellow — APPROVED** [review]
Looks good to me, but i think Vitor's opinion is more important here :)

**Comment by AndroidBob** [comment]
Build [6](https://10.10.124.134/job/wire-android-reloaded/job/PR-1438/6/) **succeeded**.

The build produced the following APK's:
- [com.wire.android-vv4.0.0(5705377)(5705377)-staging-compat.apk](https://z-lohika.s3-eu-west-1.amazonaws.com/megazord/android/reloaded/staging/compat/com.wire.android-vv4.0.0(5705377)(5705377)-staging-compat.apk)

**Pull request merged**